### PR TITLE
Remove /splits

### PR DESCRIPTION
**Pull request by severo**
The /splits processing step is redundant and uses many resources in vain. It's time to delete it:
- from the code (the workers will not be able to process these jobs, the API will not use these cache entries to serve /splits)
- from the jobs database
- from the cache database

### CODE REVIEW

**Review by albertvillanova — COMMENTED** [review]
Thanks.

Is it right that you are not removing "/splits" from:
- `openapi.json`
- `api/config.py`
- `test_11_auth.py`
- `test_12_splts.py`
- `test_21_api_metrics.py`
- `test_endpoint.py`
- `e2e/test/utils.py`
- `services/api/tests/conftest.py`
- `DEVELOPER_GUIDE.md`
- `services/api/README.md`
- `services/worker/README.md`
- `parquet.mdx`
- `quick_start.mdx`
- `server.mdx`
- `splits.mdx`

Also from docstrings of:
- `compute_dataset_split_names_response`
- `compute_dataset_split_names_from_dataset_info_response`
- `compute_dataset_split_names_from_streaming_response`

Also some nits: feel free to ignore them.

File: services/worker/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Set environment variables to configure the worker.
 - `WORKER_MAX_LOAD_PCT`: maximum load of the machine (in percentage: the max between the 1m load and the 5m load divided by the number of CPUs \*100) allowed to start a job. Set to 0 to disable the test. Defaults to 70.
 - `WORKER_MAX_MEMORY_PCT`: maximum memory (RAM + SWAP) usage of the machine (in percentage) allowed to start a job. Set to 0 to disable the test. Defaults to 80.
 - `WORKER_MAX_MISSING_HEARTBEATS`: the number of hearbeats a job must have missed to be considered a zombie job. Defaults to `5`.
-- `WORKER_ONLY_JOB_TYPES`: comma-separated list of the job types to process, e.g. "/splits,/first-rows". If empty, the worker processes all the jobs. Defaults to empty.
+- `WORKER_ONLY_JOB_TYPES`: comma-separated list of the job types to process, e.g. "/config-names,dataset-split-names". If empty, the worker processes all the jobs. Defaults to empty.
```
[the remark this answers is not in the document]
Comment (severo) [reply services/worker/README.md:20]: No, the job type is `dataset-split-names`: https://github.com/huggingface/datasets-server/blob/020d0eec506aaf0dd1650b6885c07f8b71884b77/libs/libcommon/src/libcommon/config.py#L201

`/config-names` still has to be renamed according to the new nomenclature

**Comment by severo** [comment]
> Is it right that you are not removing "/splits" from:

Yes. The string `/splits` was used for both the endpoint (still in use) and the cache entry / job type (which we are removing here).

**Review by albertvillanova — APPROVED** [review]
Thanks!

**Pull request merged**